### PR TITLE
[2.33] Fix Oauth2 token issue (#3827)

### PR DESCRIPTION
**Pull request by luciano-fiandesio**
DHIS2-7318

* fix: fix OAUTH2 bearer token retrieval

- Add "no password" encoder to security configuration to correctly
validate the "secret" for oauth2 clients
- Security configuration cleanup

* chore: added comments

(cherry picked from commit 72ce68a1e597d0ebce3139da1a38a84c74152682)

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/config/SecurityConfig.java [hunk 1]
```diff
@@ -116,33 +125,15 @@ public DefaultTokenServices defaultTokenServices( DataSource dataSource )
...
         DefaultClientDetailsService defaultClientDetailsService )
     {
-
-        DefaultTokenServices defaultTokenServices = new DefaultTokenServices();
-        defaultTokenServices.setTokenStore( new JdbcTokenStore( dataSource ) );
-        defaultTokenServices.setSupportRefreshToken( true );
-
         OAuth2AuthenticationManager oa2Manager = new OAuth2AuthenticationManager();
-        oa2Manager.setTokenServices( defaultTokenServices );
+        oa2Manager.setTokenServices( (ResourceServerTokenServices) tokenServices());
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/config/SecurityConfig.java]:
```suggestion
        oa2Manager.setTokenServices( (ResourceServerTokenServices) tokenServices() );
```

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/config/SecurityConfig.java [hunk 2]
```diff
@@ -171,14 +162,22 @@ public void configureGlobal( AuthenticationManagerBuilder auth, UserService user
...
+        auth
+            // Two factor
+            .authenticationProvider( twoFactorAuthenticationProvider )
+            // LDAP Authentication
             .authenticationProvider( customLdapAuthenticationProvider )
-            .userDetailsService( defaultClientDetailsUserDetailsService );
+            //  OAUTH2
+            .userDetailsService( defaultClientDetailsUserDetailsService )
+                // Use a non-encoding password for oauth2 secrets, since the secret is generated by the client
+                .passwordEncoder(NoOpPasswordEncoder.getInstance());
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/config/SecurityConfig.java]:
```suggestion
                .passwordEncoder( NoOpPasswordEncoder.getInstance() );
```

**Comment by luciano-fiandesio** [comment]
> Is there any simple tests we can write for this?

The only way I see it, is true an integration test.

**Comment by stian-sandvold** [comment]
> > Is there any simple tests we can write for this?
> 
> The only way I see it, is true an integration test.

Ok, I think it would be great if you could add one then, so we can get some tests for Oauth :)

**Comment by luciano-fiandesio** [comment]
> > > Is there any simple tests we can write for this?
> > 
> > 
> > The only way I see it, is true an integration test.
> 
> Ok, I think it would be great if you could add one then, so we can get some tests for Oauth :)

@vilkg will add a end 2 end test for Oauth2

**Pull request merged**